lib/AST/StmtManual.cpp

```diff
@@ -19,6 +19,95 @@ namespace pasta {
 
 #ifndef PASTA_IN_BOOTSTRAP
 
+// Follow a macro token's parent chain to the end, and returns the final parent.
+// If the macro token has no parent, this returns std::nullopt;
+std::optional<pasta::Macro>
+GetRootSubstitution(const pasta::MacroToken &macro_token) noexcept {
+  auto root = std::optional(macro_token.Parent());
```

Comment:
no need to wrap with `std::optional`, the return value should be `std::optional` already.

Reply:
Maybe have it as:

```c++
std::optional<Macro> root;
std::optional<Macro> next_root = macro_token.Parent();
do {
  root = std::move(next_root);
  next_root = root->Parent();
} while (next_root);
return root;
```

Reply:
I thought about using a `do while`, but decided to go with the `while` approach instead. I'll change it.